ext/src/main/java/org/jboss/logmanager/ext/AutoInstaller.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2019 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.logmanager.ext;
+
+import org.jboss.logmanager.LogContext;
+
+import java.util.logging.Handler;
+import java.util.logging.LogManager;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+/**
+ *
+ */
+public final class AutoInstaller {
+    private AutoInstaller() {}
+
+    /**
+     * Install the given log context into the log manager.
+     *
+     * @param logManager the log manager (must not be {@code null})
+     * @param logContext the log context (must not be {@code null})
+     * @return {@code true} if the context was installed, or {@code false} if it was already installed
+     */
+    public static boolean installInto(LogManager logManager, LogContext logContext) {
+        if (logManager instanceof org.jboss.logmanager.LogManager) {
+            return false;
+        }
+        final Logger rootLogger = logManager.getLogger("");
```

Comment:
No, in this case we're getting the LogManager that isn't ours, the one we want to hack up.

Reply:
Unless I'm missing something it should always be the `org.jboss.logmanager.LogManager` at this point. It returns `false` above if it's not.

Reply:
No, it's the opposite.  It returns `false` if it *is* ours, because then we don't need the installer.  If we get to this point it means that it's a log manager that isn't ours and we want to hijack it.